### PR TITLE
Fix bad code style found in Suggestions. 

### PR DESCRIPTION
**Pull request by budzanowski**
Fix bad code found in reviews for #8282

Use ES6 when possible.
Use functional style when possible.
Fix sloppy code.

### CODE REVIEW

**Comment by ockham** [comment]
Maybe also turn `MagicSearchWelcome` into an ES6 class (and use `localize` with it) so you get the hang of it :-)

**Comment by ockham** [comment]
> Maybe also turn `MagicSearchWelcome` into an ES6 class (and use `localize` with it) so you get the hang of it :-)

And `ThemesMagicSearchCard` :-) -- I think that will work, too, since we aren't referring to its own `refs` from outside (only inside).

We should really make that a habit -- [latest React docs don't even mention](https://facebook.github.io/react/docs/components-and-props.html) `createClass` anymore o_O

File: client/components/suggestions/index.jsx [hunk 1]
```diff
@@ -180,7 +185,7 @@ const Suggestions = React.createClass( {
 		return taxonomySuggestionsArray;
 	},
 
-	createTextWithHighlight: function( text, highlighed_text ) {
+	createTextWithHighlight( text, highlighed_text ) {
```
Comment (ockham) [inline client/components/suggestions/index.jsx]: One more nitpick -- should be camelCase: `highlightedText`

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 2]
```diff
@@ -173,31 +157,31 @@ const ThemesMagicSearchCard = React.createClass( {
...
 
 	render() {
 		const isJetpack = this.props.site && this.props.site.jetpack;
 		const isPremiumThemesEnabled = config.isEnabled( 'upgrades/premium-themes' );
 
 		const tiers = [
-			{ value: 'all', label: this.translate( 'All' ) },
-			{ value: 'free', label: this.translate( 'Free' ) },
-			{ value: 'premium', label: this.translate( 'Premium' ) },
+			{ value: 'all', label: this.props.translate( 'All' ) },
```
Comment (ockham) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Another small nit -- Might be a bit prettier if you define

```js
const translate = { this.props };
```

and use that here (and below, also for the `<Search>` `placeholder`)

File: client/my-sites/themes/themes-magic-search-card/taxonomy-styling.js [hunk 3]
```diff
@@ -1,4 +1,11 @@
 /**
+ * External dependencies
+ */
+import {
+	get
+} from 'lodash';
```
Comment (ockham) [inline client/my-sites/themes/themes-magic-search-card/taxonomy-styling.js]: Should be all on one line :-)

**Comment by ockham** [comment]
Looks very good, thanks for polishing! I left a couple of nits, but those are all non-blocking, so feel free to :ship: whenever you're happy with it!

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 4]
```diff
@@ -8,28 +8,21 @@ import classNames from 'classnames';
...
-		taxonomies: [],
-		topSearches: [],
-		suggestionsCallback: noop,
-	},
+	constructor( props ) {
+		super( props );
+	}
 
-	onMouseDown: function( event ) {
+	onMouseDown = ( event ) => {
```
Comment (gwwar) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: Interesting, is this equivalent to a bind in the constructor?

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx [hunk 5]
```diff
@@ -45,18 +38,31 @@ const MagicSearchWelcome = React.createClass( {
...
 				<div className="themes-magic-search-card__welcome-taxonomies">
 					{ this.props.taxonomies.map( taxonomy => this.renderToken( taxonomy ) ) }
 				</div>
 			</div>
 		);
 	}
-} );
+}
+
+MagicSearchWelcome.propTypes = {
```
Comment (gwwar) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: You can define these as static properties in the class:
```
class MagicSearchWelcome extends React.Component {

  static propTypes = {
    //...
  };

  static defaultProps = {
    //...
  };
```

File: client/my-sites/themes/themes-magic-search-card/welcome.jsx (hunk 4 above)
Comment (dmsnell) [inline client/my-sites/themes/themes-magic-search-card/welcome.jsx]: They are class properties and yep we are using them.

File: client/my-sites/themes/themes-magic-search-card/index.jsx (hunk 2 above)
Comment (budzanowski) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: :+1:

**Pull request merged**